Reply/app/src/main/java/com/example/reply/ui/ReplyListContent.kt

```diff
@@ -113,20 +119,17 @@ fun ReplyInboxScreen(
             )
             // When we have bottom navigation we show FAB at the bottom end.
             if (navigationType == ReplyNavigationType.BOTTOM_NAVIGATION) {
-                LargeFloatingActionButton(
+                ExtendedFloatingActionButton (
+                    text = { Text(text = stringResource(id = R.string.compose)) },
+                    icon = { Icon(Icons.Default.Edit, stringResource(id = R.string.compose)) },
                     onClick = { /*TODO*/ },
                     modifier = Modifier
                         .align(Alignment.BottomEnd)
                         .padding(16.dp),
                     containerColor = MaterialTheme.colorScheme.tertiaryContainer,
-                    contentColor = MaterialTheme.colorScheme.onTertiaryContainer
-                ) {
-                    Icon(
-                        imageVector = Icons.Default.Edit,
-                        contentDescription = stringResource(id = R.string.compose),
-                        modifier = Modifier.size(28.dp)
-                    )
-                }
+                    contentColor = MaterialTheme.colorScheme.onTertiaryContainer,
+                    expanded = emailLazyListState.isScrollingUp()
```

Comment:
No need for an extension here, you can replace this with
```suggestion
                    expanded = emailLazyListState.lastScrolledBackward ||
                            !emailLazyListState.canScrollBackward
```

Reply:
Thank you! I really appreciate it. It's been exciting to try Android development again after 4 years!